src/gsc_sql.c

```diff
@@ -0,0 +1,379 @@
+    }
+
+    int fieldCount = mysql_num_fields((MYSQL_RES *) result);
+    Scr_AddInt( fieldCount );
+}
+
+/* =================================================================
+* URL
+    http://dev.mysql.com/doc/refman/5.7/en/mysql-fetch-row.html
+
+* Description
+    Retrieves the next row of a result set.
+
+* Return Value(s)
+    A MYSQL_ROW structure for the next row. NULL if there are no more rows to retrieve or if an error occurred.
+   ================================================================= */
+void gsc_mysql_fetch_rows() {
+    int results = Scr_GetInt(0);
+    MYSQL_RES* result = (MYSQL_RES*) results;
+
+    if (Scr_GetNumParam() != 1) {
+        Scr_Error("Usage: gsc_mysql_fetch_row(query result from gsc_mysql_query);\n");
+    }
+
+    if( mysql_num_rows(result) != 0 )
+    {
+        int num_fields = mysql_num_fields(result);
+        MYSQL_FIELD *field;
+        MYSQL_ROW row;
+        MYSQL_FIELD* fieldArray[num_fields];
+        int count = 0;
+        Scr_MakeArray();
+
+        /* Check to see if this is a single element row */
+        if (mysql_num_rows(result) == 1) {
+            while((field = mysql_fetch_field(result))) {
+                fieldArray[count] = field;
+                count++;
+            }
+            row = mysql_fetch_row(result);
+            Scr_MakeArray();
+            for (int i = 0; i < num_fields; i++) {
+                Scr_AddString(row[i]);
+                int tempId = Scr_AllocString(fieldArray[i]->name);
+                Scr_AddArrayKey(tempId);
+            }
+            return;
+        }
+
+        /* If it's not a single row column, get row data*/
+        while((row = mysql_fetch_row(result))){
+            Scr_MakeArray();
+            while((field = mysql_fetch_field(result))) {
+                fieldArray[count] = field;
+                count++;
+            }
+
+            for (int i = 0; i < num_fields; i++) {
+                Scr_AddString(row[i]);
+                int tempId = Scr_AllocString(fieldArray[i]->name);
+                Scr_AddArrayKey(tempId);
+            }
+            Scr_AddArray();
+        }
+    }
+}
+
+/* =================================================================
+* URL
+    http://dev.mysql.com/doc/refman/5.7/en/mysql-errno.html
+
+* Description
+    For the connection specified by mysql, mysql_errno() returns the
+    error code for the most recently invoked API function that can
+    succeed or fail. A return value of zero means that no error occurred.
+    Client error message numbers are listed in the MySQL errmsg.h header
+    file.  Server error message numbers are listed in mysqld_error.h.
+    Errors also are listed at Appendix B, Errors, Error Codes, and
+    Common Problems.
+
+* Return Value(s)
+    An error code value for the last mysql_xxx() call, if it failed. Zero means no error occurred.
+   ================================================================= */
+void gsc_mysql_errno() {
```

Comment:
@T-Maxxx all the functions will work or return undefined. Its up to the devs to use that to there advantage. The errno and error functions are for finding things that don't work but should work. Say it wasn't able to INSERT because of permission issues. The error function work fix that. Printing to console is also a good idea and thats why we've implemented `Com_DPrintf` into some of the functions so that if your running developer you will see the errors. Of course no one actually uses developer because that would mean they would have to develop their mods and maps correctly and well.. this is call of duty.. nobody does that. So thats their fault.